draft-ietf-httpbis-client-hints.md

```diff
@@ -190,6 +200,12 @@ Implementers ought to consider both user and server controlled mechanisms and po
 Implementers SHOULD support Client Hints opt-in mechanisms and MUST clear persisted opt-in preferences when any one of site data, browsing history, browsing cache, or similar, are cleared.
 
 
+## Abuse Detection
+A user agent that tracks access to active fingerprinting information SHOULD consider emission of Client Hints headers similarly to the way it would consider access to the equivalent API.
+
+User agents which detect abuse through forensic means can observe HTTP responses to requests containing Client Hints, and their variance based on those hints. Origins which opt-in to Client Hints but don't vary any of their responses accordingly are likely using that information for nefarious purposes.
```

Comment:
This last sentence is a little tricky. I don't think we have any data to support the 'likely' claim - we're assuming that, but it may be that it turns out folks just have trouble with CH and implementing it (in)correctly.


> As Client Hints are provided in order to allow variation and optimization of the content to the provided Client-Hints, resource responses that do not vary on the Client Hints provided may indicate that such Hints are being used for nefarious purposes.

I mean, it makes it weaker in terms of *is* being used nefariously, so perhaps that's not worth it. I'm just a little uncomfortable stating such things without data, and without explanation about why the conclusion is there.

Reply:
yeah, that's fair